config/3.3/eventManager/templates/gobal-image-pull-secret.yaml

```diff
@@ -0,0 +1,112 @@
+
+              if [ "$CP_STG_KEY" == "" ]; then
+                  echo "Need image pull secret for internal build: CP_STG_KEY"
+                  exit 1
+              fi
+
+              CP_STG_USER="${CP_STG_USER:=iamapikey}"
+              CP_STG_AUTH="$CP_STG_USER:$CP_STG_KEY"
+
+              # Extract the existing pull secret from cluster
+              rm -rf ${HOME}/.dockerconfigjson
+              oc extract secret/pull-secret -n openshift-config --to ${HOME} --confirm
+              if [[ $? -ne 0 ]]; then
+                  logger "[ERROR] Failed to extract existing pull-secret!"
+                  return 1
+              fi
+
+              oc registry login --registry cp.stg.icr.io \
+              --auth-basic=$CP_STG_AUTH \
+              --to=${HOME}/.dockerconfigjson
+
+              # Apply the new secret
+              oc set data secret/pull-secret --from-file .dockerconfigjson=${HOME}/.dockerconfigjson -n openshift-config
+
+              wait_time=40
+              # master nodes
+              sleep $wait_time
+              num_masters=$(oc get machineconfigpools master --no-headers -o custom-columns=":status.machineCount")
+              num_masters_ready=$(oc get machineconfigpools master --no-headers -o custom-columns=":status.readyMachineCount")
+              # worker nodes
+              num_workers=$(oc get machineconfigpools worker --no-headers -o custom-columns=":status.machineCount")
+              num_workers_ready=$(oc get machineconfigpools worker --no-headers -o custom-columns=":status.readyMachineCount")
+              attempt=1
+              if [ $num_masters_ready -lt $num_masters ]; then
+                while [[ $num_masters_ready -lt $num_masters ]] && [[ $attempt -le $attempts ]]
+                do
+                  echo "Not all master nodes updated"
+                  echo "$num_masters_ready/$num_masters master nodes currently ready - $attempt/$attempts"
+                  ((attempt=$attempt+1))
+                  sleep $wait_time
+                  num_masters_ready=$(oc get machineconfigpools master --no-headers -o custom-columns=":status.readyMachineCount")
+                done
+              else
+                echo "All master nodes updated"
+              fi
+              attempt=1
+              if [ $num_workers_ready -lt $num_workers ]; then
+                while [[ $num_workers_ready -lt $num_workers ]] && [[ $attempt -le $attempts ]]
+                do
+                  echo "Not all worker nodes updated"
+                  echo "$num_workers_ready/$num_workers worker nodes currently ready - $attempt/$attempts"
+                  ((attempt=$attempt+1))
+                  sleep $wait_time
+                  num_workers_ready=$(oc get machineconfigpools worker --no-headers -o custom-columns=":status.readyMachineCount")
+                done
+              else
+                echo "All worker nodes updated"
+              fi
+
+              cat <<EOF | kubectl apply -f -
```

Comment:
Why AI Manager does not need this?

Reply:
Only the image of eventmanager needs to be redirected,  the image from AIManager all points directly to the correct address.

Reply:
Only the image of eventmanager needs to be redirected,  the images from AIManager all points directly to the correct address